homeassistant/components/rest/__init__.py

```diff
@@ -149,8 +149,12 @@ def _rest_coordinator(
     """Wrap a DataUpdateCoordinator around the rest object."""
     if resource_template:
 
-        async def _async_refresh_with_resource_template():
-            rest.set_url(resource_template.async_render(parse_result=False))
+        async def _async_refresh_with_resource_template() -> None:
+            rest.set_url(
+                cast(template.Template, resource_template).async_render(
+                    parse_result=False
+                )
+            )
```

Comment:
It seems that mypy treats resource_template as `template.Template | None`
Another option is using TYPE_CHECKING but that is also frowned upon.

```python
        async def _async_refresh_with_resource_template() -> None:
            if TYPE_CHECKING:
                assert resource_template
            rest.set_url(resource_template.async_render(parse_result=False))
            await rest.async_update()
```